test/spiretest/socketapi_windows.go

```diff
@@ -18,6 +18,8 @@ import (
 	"google.golang.org/grpc"
 )
 
+var rnd = rand.New(rand.NewSource(time.Now().UnixNano())) // nolint: gosec // used for testing only
```

Comment:
The functions that use the package-level rand source are safe to call concurrently but those created with rand.New are not. Though our current use is probably all from a single goroutine, if that changes, this will have a data race. We should probably protect this with a mutex.

Reply:
Yeah, makes sense. I added a package-level mutex and wrapped the call to `Uint64()` with a function that grabs the lock before accessing the local `*rand.Rand`.